File: webclient/public/outcome.js
class Outcome extends Component {

    constructor(props) {
        super(props);

        this.onclick = this.onclick.bind(this);
    }
    state = {
        Clicked: false
    }
    getPrice = () => {
        return this.props.Outcome.TrueProbability * 1.05
    }

    formatText = () => {
        var outcomeName = this.props.Outcome.HostType
        switch (this.props.Outcome.OutcomeType) {
            case "Yes / No":
                outcomeName = "Yes"
                if (this.props.Outcome.HostTypes == "Away") {
                    outcomeName = "No"
                }
                case "Over / Under":
                    outcomeName = "Over"
                    if (this.props.Outcome.HostTypes == "Away") {
                        outcomeName = "Under"
                    }
        }
        return outcomeName
    }

    onclick = (e) => {
        e.preventDefault();
        this.setState({
            Clicked: !this.state.Clicked
        })
    }

    render = () => {
        var betPlacer = < span > < /span>
        if (this.state.Clicked) {
            betPlacer = < BetPlacer Price = {
                this.getPrice()
            }
            />
        }

        return ( <
            div className = "outcome" >
            <
            a onClick = {
                this.onclick
            }
            href = "#" >
            <
            p > < b > {
                this.formatText()
            } < /b></p >
            <
            p > {
                this.getPrice()
            } < /p> <
            /a> {
                betPlacer
            } <
            /div>
        )
    }
}

File: model/calculator/Outcome.go
package Calculator

import (
	"cointossbookie/ScoutInfo"
	"fmt"
)

type Outcome struct {
	TrueProbability float32
	HostType        ScoutInfo.HostType
	OutcomeType     OutcomeType
}

func NewOutcomes(home float32, away float32, outcomeType OutcomeType) []Outcome {
	var outcomes = []Outcome{
		Outcome{home, ScoutInfo.HostTypeHome, outcomeType},
		Outcome{away, ScoutInfo.HostTypeAway, outcomeType},
	}
	return outcomes
}

func (s Outcome) GetPrice() float32 {
	return s.TrueProbability * 1.05
}

func (s Outcome) String() string {
	outcomeName := s.HostType.String()

	switch s.OutcomeType {
	case OutcomeTypeYesNo:
		outcomeName = "Yes"
		if s.HostType == ScoutInfo.HostTypeAway {
			outcomeName = "No"
		}
	case OutcomeTypeOverUnder:
		outcomeName = "Over"
		if s.HostType == ScoutInfo.HostTypeAway {
			outcomeName = "Under"
		}
	}
	return fmt.Sprintf("%s: %.2f (TP: %.2f)", outcomeName, s.GetPrice(), s.TrueProbability)
}
